src/core/inc/EnvironmentOptions.h

```diff
@@ -94,6 +94,15 @@ class EnvOptionsValues
   //! Copy constructor
   EnvOptionsValues(const EnvOptionsValues& src);
 
+  //! Set parameter option names to begin with prefix
+  void set_prefix(const char* prefix);
+
+  //! Set default values for parameter options
+  void set_defaults();
+
+  //! Given prefix, read the input file for parameters named "prefix"+*
+  void parse(const BaseEnvironment* env, const char* prefix);
```

Comment:
I kept pointers such as const char* and Env* for consistency with ctor instead of changing APIs to use std::string and Env&

Reply:
I wouldn't worry about consistency in a brand-new API, and I wouldn't worry too much about stylistic consistency in QUESO.

Using a reference is IMHO always better than a pointer to a class when you expect the pointer to never be null.

Using a char pointer rather than converting to std::string is often a nice way to buy a bit of efficiency at the expense of a bit of safety, but the former is negligible in a method that'll only be called a handful of times at most per execution, so I'd prefer the latter.